carbonmark/hooks/useFetchUser.ts

```diff
@@ -1,21 +1,21 @@
+import { ClientT, client } from "lib/api/client";
 import { urls } from "lib/constants";
-import { fetcher } from "lib/fetcher";
-import { User } from "lib/types/carbonmark.types";
 import type { SWRConfiguration } from "swr";
 import useSWR from "swr";
 
-type Opts = SWRConfiguration & {
-  network?: "mumbai" | "polygon";
-};
+type GetUserFnT = ClientT["/users/{walletOrHandle}"]["get"];
+
+export const useFetchUser = (
+  args: Parameters<GetUserFnT>[0],
+  options?: SWRConfiguration
+) => {
+  const fetchUser = async () =>
+    (await client["/users/{walletOrHandle}"].get(args)).json();
 
-export const useFetchUser = (address?: string, options?: Opts) => {
-  const { network = "polygon", ...swrConfig } = options || {};
-  const { data, ...rest } = useSWR<User>(
-    address
-      ? `${urls.api.users}/${address}?type=wallet&network=${network}`
-      : null,
-    fetcher,
-    swrConfig
+  const { data, ...rest } = useSWR(
+    `${urls.api.users}/${args.params.walletOrHandle}&network=${args.query?.network}`,
```

Comment:
This string is used by SWR as a key to track changes, so we need to:
* [ ] make sure the key changes only when a param changes
* [ ] make sure the fallback key matches the one provided to the SWRConfig in `components/pages/Users/index.tsx`. (Create a shared util to construct both?)
* [ ] make sure address is always lower case (I noticed on production that we aren't consistent here)

Reply:
Added a temporary solution to this via a key gen function. A more solid solution is in the works.